src/shared/range-input-inner-container/range-input-inner-container.styles.tsx

```diff
@@ -0,0 +1,110 @@
+import { ArrowRightIcon } from "@lifesg/react-icons";
+import styled, { css } from "styled-components";
+import { Color } from "../../color";
+
+// =============================================================================
+// STYLE INTERFACE
+// =============================================================================
+interface GeneralStyleProps {
+    $minWidthBeforeWrap?: number;
+}
+
+interface IndicatorStyleProps extends GeneralStyleProps {
+    $position: "start" | "end" | "none";
+    $error: boolean;
+}
+
+// =============================================================================
+// STYLING
+// =============================================================================
+
+const getDynamicMediaWidthQuery = (width: number) => {
+    return `@media screen and (max-width: ${width}px)`;
+};
+
+export const Wrapper = styled.div<GeneralStyleProps>`
+    position: relative;
+    height: 100%;
+    display: flex;
+    flex: 1;
+
+    ${(props) => {
+        if (props.$minWidthBeforeWrap) {
+            return css`
+                ${getDynamicMediaWidthQuery(props.$minWidthBeforeWrap)} {
+                    /* Parent container need to provide space */
+                    flex-wrap: wrap;
+
+                    [data-id="range-container-elem1-container"],
+                    [data-id="range-container-elem2-container"] {
+                        // 100% - Icon size - 2padding
+                        max-width: calc(100% - 1.125rem - 1rem);
```

Comment:
this seems to only work when `Wrapper` has set an explicit width like `width: 100%`

Reply:
Discussed to leave style as it is and amend accordingly when used with `DateRangeInput`